Assets/MixedRealityToolkit.Providers/ObjectMeshObserver/SpatialObjectMeshObserver.cs

```diff
@@ -0,0 +1,336 @@
+                        meshFilters[i].sharedMesh,
+                        MeshPhysicsLayer,
+                        $"Spatial Object Mesh {currentMeshId}",
+                        currentMeshId,
+                        ObservedObjectParent);
+
+                    meshObject.GameObject.transform.localPosition = meshFilters[i].transform.position;
+                    meshObject.GameObject.transform.localRotation = meshFilters[i].transform.rotation;
+
+                    ApplyMeshMaterial(meshObject);
+
+                    meshes.Add(currentMeshId, meshObject);
+
+                    currentMeshId++;
+                }
+            }
+            sendObservations = false;
+        }
+
+        /// <inheritdoc />
+        public override void Suspend()
+        {
+            IsRunning = false;
+        }
+
+        private void RemoveMeshObject(int id)
+        {
+            SpatialAwarenessMeshObject meshObject = null;
+
+            if (meshes.TryGetValue(id, out meshObject))
+            {
+                // Remove the mesh object from the collection.
+                meshes.Remove(id);
+                if (meshObject != null)
+                {
+                    SpatialAwarenessMeshObject.Cleanup(meshObject);
+                }
+
+                // Send the mesh removed event
+                SpatialAwarenessSystem?.RaiseMeshRemoved(this, id);
+            }
+        }
+
+        #endregion IMixedRealitySpatialAwarenessObserver implementation
+
+        #region IMixedRealitySpatialAwarenessMeshObserver implementation
+
+        public SpatialAwarenessMeshDisplayOptions displayOption = SpatialAwarenessMeshDisplayOptions.Visible;
+        
+        /// <inheritdoc />
+        public SpatialAwarenessMeshDisplayOptions DisplayOption
+        {
+            get => displayOption;
+
+            set
+            {
+                displayOption = value;
+                ApplyUpdatedMeshDisplayOption(displayOption);
+            }
+        }
+
+        /// <inheritdoc />
+        public SpatialAwarenessMeshLevelOfDetail LevelOfDetail { get; set; } = SpatialAwarenessMeshLevelOfDetail.Coarse;
+
+        private Dictionary<int, SpatialAwarenessMeshObject> meshes = new Dictionary<int, SpatialAwarenessMeshObject>();
+
+        /// <inheritdoc />
+        public IReadOnlyDictionary<int, SpatialAwarenessMeshObject> Meshes =>
+            new Dictionary<int, SpatialAwarenessMeshObject>(meshes) as IReadOnlyDictionary<int, SpatialAwarenessMeshObject>;
+
+        private int meshPhysicsLayer = 31;
+
+        /// <inheritdoc />
+        public int MeshPhysicsLayer
+        {
+            get => meshPhysicsLayer;
+
+            set
+            {
+                if ((value < 0) || (value > 31))
+                {
+                    Debug.LogError("Specified MeshPhysicsLayer is out of bounds. Please use a value between 0 and 31, inclusive.");
+                    return;
+                }
+                meshPhysicsLayer = value;
```

Comment:
If this property is changed at runtime, at what point does an already loaded mesh get moved to the updated physics layer? It appears the code in **Resume** does not run continuously, and that appears to be the only time the layer is used.

Reply:
interesting.. i dont think even the live mesh handles that :)

Reply:
added support and confirmed in editor.